### PR TITLE
feat(nuxt): smooth scroll behavior option

### LINKED ISSUE #15728: Set smooth scroll on hash links.
### Describe the feature

The default router behavior is great except for one small thing. For hash links, I would like to be able to use smooth scrolling. Perhaps this could be an option in nuxt.config?

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

### PR DESCRIPTION
**Pull request by lvpaspaltzis**
### 🔗 Linked issue

resolves #15728

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add the `smoothScrollBehavior` option to `RouterOptions` that allows smooth scroll for hash links

https://github.com/nuxt/nuxt/assets/22207414/4f12d850-7468-4091-bdbf-4e7ca38d0294

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by vargasmesh** [comment]
Hi @danielroe. I have some questions regarding this change:
1. Can't we add this option `RouterConfigSerializable` so there is no need to create an `app/router.options.ts` and use it directly on `nuxt.config.ts`?
2. Should we update https://nuxt.com/docs/guide/going-further/custom-routing#router-options with this new option?

File: packages/schema/src/types/router.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ export type RouterOptions = Partial<Omit<_RouterOptions, 'history' | 'routes'>>
   history?: (baseURL?: string) => RouterHistory
   routes?: (_routes: _RouterOptions['routes']) => _RouterOptions['routes']
   hashMode?: boolean
+  smoothScrollBehavior?: boolean
```
Comment (danielroe) [inline packages/schema/src/types/router.ts:8]: what about:

```suggestion
  scrollBehaviorType?: 'smooth' | 'auto'
```

Comment (vargasmesh) [reply packages/schema/src/types/router.ts]: Done!

**Review by danielroe — APPROVED** [review]

**Comment by mkarras** [comment]
'scroll-behavior: smooth;' via CSS should do the same.

**Pull request merged**